app/models/list_import/anime_planet.rb

```diff
@@ -0,0 +1,82 @@
+# rubocop:disable Metrics/LineLength
+# == Schema Information
+#
+# Table name: list_imports
+#
+#  id                      :integer          not null, primary key
+#  error_message           :text
+#  error_trace             :text
+#  input_file_content_type :string
+#  input_file_file_name    :string
+#  input_file_file_size    :integer
+#  input_file_updated_at   :datetime
+#  input_text              :text
+#  progress                :integer
+#  status                  :integer          default(0), not null
+#  strategy                :integer          not null
+#  total                   :integer
+#  type                    :string           not null
+#  created_at              :datetime         not null
+#  updated_at              :datetime         not null
+#  user_id                 :integer          not null
+#
+# rubocop:enable Metrics/LineLength
+
+class ListImport
+  class AnimePlanet < ListImport
+    ANIME_PLANET_HOST = 'http://www.anime-planet.com/users/'.freeze
+
+    # accepts a username as input
+    validates :input_text, length: {
+      minimum: 3,
+      maximum: 20
+    }, presence: true
+    # does not accept file uploads
```

Comment:
You should validate the absence of the `input_file` even though you can't easily test it.